src/Orchard.Web/Modules/Orchard.DynamicForms/Drivers/QueryElementDriver.cs

```diff
@@ -174,5 +174,29 @@ public QueryElementDriver(IFormsBasedElementServices formsServices, IProjectionM
             var runtimeValue = element.RuntimeValue;
             return runtimeValue != null ? runtimeValue.Split(new[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries) : Enumerable.Empty<string>();
         }
+        
+        protected override void OnExporting(Query element, ExportElementContext context) {
+            base.OnExporting(element, context);
+            if (element.QueryId == null)
+                return;
+            var queryContentItem = _contentManager.Get<QueryPart>(element.QueryId.Value);
+            if (queryContentItem == null)
+                return;
+            context.Element.Data.Add("QueryElementIdentity", _contentManager.GetItemMetadata(queryContentItem).Identity.ToString());
+        }
+
+        protected override void OnImported(Query element, ImportElementContext context) {
+            base.OnImported(element, context);
+            if (!context.Element.Data.Keys.Contains("QueryElementIdentity"))
+                return;
+            var queryIdentifier = context.Element.Data["QueryElementIdentity"];
+            var queryContentItem = context.Session.GetItemFromSession(queryIdentity).Content;
+            if (queryIdentityPart == null)
```

Comment:
queryIdentityPart is not defined either

Reply:
arrh, it's my fault. Just edit file in Github editor. I will fix it.

And I think if I should override the OnImportCompleted method. It seems that it is more fit for this scenario

Reply:
@li0803 will you provide a PR?

Reply:
Just fixed this. But seems there is conflict and I need to merge dev to my branch and there are huge changes to be merged as my original branch is based on 1.10.2.  :(
any sugesstion